core/web/resolver/eth_transaction.go

```diff
@@ -0,0 +1,144 @@
+package resolver
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager"
+	"github.com/smartcontractkit/chainlink/core/utils/stringutils"
+	"github.com/smartcontractkit/chainlink/core/web/loader"
+)
+
+type EthTransactionResolver struct {
+	tx bulletprooftxmanager.EthTx
+}
+
+func NewEthTransaction(tx bulletprooftxmanager.EthTx) *EthTransactionResolver {
+	return &EthTransactionResolver{tx: tx}
+}
+
+func NewEthTransactions(results []bulletprooftxmanager.EthTx) []*EthTransactionResolver {
+	var resolver []*EthTransactionResolver
+
+	for _, tx := range results {
+		resolver = append(resolver, NewEthTransaction(tx))
+	}
+
+	return resolver
+}
+
+func (r *EthTransactionResolver) State() string {
+	return string(r.tx.State)
+}
+
+func (r *EthTransactionResolver) Data() hexutil.Bytes {
+	return hexutil.Bytes(r.tx.EncodedPayload)
+}
+
+func (r *EthTransactionResolver) From() string {
+	return r.tx.FromAddress.String()
+}
+
+func (r *EthTransactionResolver) To() string {
+	return r.tx.ToAddress.String()
+}
+
+func (r *EthTransactionResolver) GasLimit() string {
+	return stringutils.FromInt64(int64(r.tx.GasLimit))
+}
+
+func (r *EthTransactionResolver) GasPrice() string {
+	return r.tx.EthTxAttempts[0].GasPrice.String()
+}
+
+func (r *EthTransactionResolver) Value() string {
+	return r.tx.Value.String()
+}
+
+func (r *EthTransactionResolver) EVMChainID() graphql.ID {
+	return graphql.ID(r.tx.EVMChainID.String())
+}
+
+func (r *EthTransactionResolver) Nonce() *string {
+	if r.tx.Nonce == nil {
+		return nil
+	}
+
+	value := stringutils.FromInt64(*r.tx.Nonce)
+
+	return &value
+}
+
+func (r *EthTransactionResolver) Hash() string {
+	return r.tx.EthTxAttempts[0].Hash.Hex()
+}
+
+func (r *EthTransactionResolver) Hex() string {
+	return hexutil.Encode(r.tx.EthTxAttempts[0].SignedRawTx)
+}
+
+// Chain resolves the node's chain object field.
+func (r *EthTransactionResolver) Chain(ctx context.Context) (*ChainResolver, error) {
+	chain, err := loader.GetChainByID(ctx, string(r.EVMChainID()))
+	if err != nil {
+		return nil, err
+	}
+
+	return NewChain(*chain), nil
+}
+
+func (r *EthTransactionResolver) Attempts() []*EthTransactionAttemptResolver {
+	return NewEthTransactionsAttempts(r.tx.EthTxAttempts)
```

Comment:
It would be better to dataload this to reduce database load

Reply:
implemented